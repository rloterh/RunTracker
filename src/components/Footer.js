import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Footer = ({
  home, feed, progress, settings,
}) => (
  <footer className="bg-footer bg-opacity-80 bottom-0 fixed w-full flex justify-between shadow-md border-t border-blue-900">
    <div className="border-r w-1/4 flex items-center justify-center py-3">
      <Link to={home}>
        <svg className="h-8" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="#fff" stroke="#fff" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
          <line x1="18" y1="20" x2="18" y2="10" />
          <line x1="12" y1="20" x2="12" y2="4" />
          <line x1="6" y1="20" x2="6" y2="14" />
        </svg>
      </Link>
    </div>
    <div className="w-1/4 flex items-center justify-center border-r">
      <Link to={feed}>
        <svg className="h-8" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="#fff" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
          <polyline points="23 6 13.5 15.5 8.5 10.5 1 18" />
          <polyline points="17 6 23 6 23 12" />
        </svg>
      </Link>
    </div>
    <div className="w-1/4 flex items-center justify-center border-r">
      <Link to={progress}>
        <svg className="h-6" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="#fff" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
          <path d="M21.21 15.89A10 10 0 1 1 8 2.83" />
          <path d="M22 12A10 10 0 0 0 12 2v10z" />
        </svg>
      </Link>
    </div>
    <div className="w-1/4 flex items-center justify-center">
      <Link to={settings}>
        <svg xmlns="http://www.w3.org/2000/svg" className="h-8" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="#fff" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
          <circle cx="12" cy="12" r="1" />
          <circle cx="19" cy="12" r="1" />
          <circle cx="5" cy="12" r="1" />
        </svg>
      </Link>
    </div>
  </footer>
);

export default Footer;

Footer.defaultProps = {
  home: '',
  feed: '',
  progress: '',
  settings: '',
};

Footer.propTypes = {
  home: PropTypes.string,
  feed: PropTypes.string,
  progress: PropTypes.string,
  settings: PropTypes.string,
};
