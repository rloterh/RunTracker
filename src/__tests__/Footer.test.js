import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Footer from '../components/Footer';
import store from '../redux/configureStore';

describe('checks if Home component renders correctly', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Footer />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
