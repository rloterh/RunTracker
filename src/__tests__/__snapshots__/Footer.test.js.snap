// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Home component renders correctly should render correctly 1`] = `
<footer
  className="bg-footer bg-opacity-80 bottom-0 fixed w-full flex justify-between shadow-md border-t border-blue-900"
>
  <div
    className="border-r w-1/4 flex items-center justify-center py-3"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        className="h-8"
        fill="#fff"
        height="24"
        stroke="#fff"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="18"
          x2="18"
          y1="20"
          y2="10"
        />
        <line
          x1="12"
          x2="12"
          y1="20"
          y2="4"
        />
        <line
          x1="6"
          x2="6"
          y1="20"
          y2="14"
        />
      </svg>
    </a>
  </div>
  <div
    className="w-1/4 flex items-center justify-center border-r"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        className="h-8"
        fill="none"
        height="24"
        stroke="#fff"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="23 6 13.5 15.5 8.5 10.5 1 18"
        />
        <polyline
          points="17 6 23 6 23 12"
        />
      </svg>
    </a>
  </div>
  <div
    className="w-1/4 flex items-center justify-center border-r"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        className="h-6"
        fill="none"
        height="24"
        stroke="#fff"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21.21 15.89A10 10 0 1 1 8 2.83"
        />
        <path
          d="M22 12A10 10 0 0 0 12 2v10z"
        />
      </svg>
    </a>
  </div>
  <div
    className="w-1/4 flex items-center justify-center"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        className="h-8"
        fill="none"
        height="24"
        stroke="#fff"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="1"
        />
        <circle
          cx="19"
          cy="12"
          r="1"
        />
        <circle
          cx="5"
          cy="12"
          r="1"
        />
      </svg>
    </a>
  </div>
</footer>
`;
